### ISSUE TITLE
Suggested improvements: Nth-Prime

### ISSUE THREAD

**Issue by gavinhenderson** [exercism/javascript#540]
I have a few questions/suggestions to improve the 'nth-prime' challenge. 

Should we make people use the [Prime Number Theorem](https://en.wikipedia.org/wiki/Prime_number_theorem)? Right now people can write a solution that just finds a preset amount of primes and then picks the nth prime (which is what the example does). This is obviously not an optimized approach as you will be waiting the same time to get the 1st prime as the 100th. Could this be solved by checking the time taken to find 100 primes is greater than that of 1 in a test?

If we were to add the test: 
```js
it('huge prime', function () {
    expect(prime.nth(18000)).toEqual(200183);
  });
```
then the example code would fail as not enough primes are generated. To me the example code isn't that robust and the tests allow for code that isn't robust as well.

Edit: I also thought that this currently allows the exact copying of the previous 'sieve' exercise, with a small addition of grabbing a single value from the array it gives. If this can be solved with almost no unique code then is it doing its job?

**Comment by gavinhenderson** [comment]
@kytrinyx Is this something that this an issue shared by others? If so I happy to open a PR?

**Comment by matthewmorgan** [comment]
@gavinhenderson I appreciate your thoughts on this.  

I think there are three issues you raise here, and I'm unconcerned about the last (unique code from the sieve exercise.) 

I don't think we should require anyone to implement their solution with any specific algorithm-- that's really what the peer-to-peer discussion is about.

However, I think a better test for a quality method of generating primes would be welcome!  I encourage you to submit a PR, let's see how long it takes to build-- is this practical?  I'd love to know.

Please let me know if I can assist further.

**Comment by matthewmorgan** [comment]
@gavinhenderson By 'unconcerned,' I just mean that I think it's OK for people to realize that they can leverage code that they've already written.  I do that every day :)  I also think it's OK for them to decide to write new code, and I think both choices can provoke discussions like the one we're having now.

I also understand that some people will read the example code in an attempt to expand their understanding of what can and should be done, so I think it's great that you are willing to help us upgrade the example solution code.

I look forward to reviewing the PR!

**Comment by gavinhenderson** [comment]
Fixed #542
